### PR TITLE
Datum rev & related upgrades to base datum antag

### PR DESCRIPTION
**Pull request by AnturK**
Pending:
- [x] Cleanup
- [x] Testing
- [x] More cleanup

### CODE REVIEW

**Comment by MoreRobustThanYou** [comment]
standardize datum antag file naming reeee!

File: code/datums/antagonists/revolution.dm [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+/datum/antagonist/rev/proc/promote()
+	var/old_team = rev_team
+	var/datum/mind/old_owner = owner
+	silent = TRUE
+	owner.remove_antag_datum(/datum/antagonist/rev)
+	var/datum/antagonist/rev/head/new_revhead = new(old_owner)
+	new_revhead.silent = TRUE
+	old_owner.add_antag_datum(new_revhead,old_team)
+	new_revhead.silent = FALSE
+	to_chat(old_owner, "<span class='userdanger'>You have proved your devotion to revolution! Yea are a head revolutionary now!</span>")
```
Comment (JJRcop) [inline code/datums/antagonists/revolution.dm]: nitpick: "Yea are" doesn't sound quite right...

**Comment by praisenarsie** [comment]
antur is on a roll with the datum antags. is he currently sane?

File: code/datums/antagonists/revolution.dm [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+		return
+	. = ..()
+
+/datum/antagonist/rev/head/equip_rev()
+	var/mob/living/carbon/human/H = owner.current
+	if(!istype(H))
+		return
+
+	if (owner.assigned_role == "Clown")
+		to_chat(owner, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
```
Comment (JJRcop) [inline code/datums/antagonists/revolution.dm]: nitpick #2: This fluff won't line up with a clown being promoted to head rev later in the round

File: code/datums/mind.dm [hunk 3]
```diff
@@ -932,57 +930,40 @@
...
 				message_admins("[key_name_admin(usr)] has de-rev'ed [current].")
 				log_admin("[key_name(usr)] has de-rev'ed [current].")
 			if("rev")
-				if(src in SSticker.mode.head_revolutionaries)
-					SSticker.mode.head_revolutionaries -= src
-					SSticker.mode.update_rev_icons_removed(src)
-					to_chat(current, "<span class='userdanger'>Revolution has been disappointed of your leader traits! You are a regular revolutionary now!</span>")
-				else if(!(src in SSticker.mode.revolutionaries))
-					to_chat(current, "<span class='danger'><FONT size = 3> You are now a revolutionary! Help your cause. Do not harm your fellow freedom fighters. You can identify your comrades by the red \"R\" icons, and your leaders by the blue \"R\" icons. Help them kill the heads to win the revolution!</FONT></span>")
+				if(has_antag_datum(/datum/antagonist/rev/head))
```
Comment (JJRcop) [inline code/datums/mind.dm]: minor, trivial: Can condense these three has_antag_datum calls into 1 by declaring ```head``` before this and checking head

File: code/datums/antagonists/revolution.dm [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+/datum/objective_team/revolution/proc/update_heads()
+	if(SSticker.HasRoundStarted())
+		var/list/datum/mind/head_revolutionaries = head_revolutionaries()
+		var/list/datum/mind/heads = SSjob.get_all_heads()
+		var/list/sec = SSjob.get_all_sec()
+
+		if(head_revolutionaries.len < max_headrevs && head_revolutionaries.len < round(heads.len - ((8 - sec.len) / 3)))
+			var/list/datum/mind/non_heads = members - head_revolutionaries
+			var/list/datum/mind/promotable = list()
+			for(var/datum/mind/khrushchev in non_heads)
```
Comment (JJRcop) [inline code/datums/antagonists/revolution.dm]: 😄

**Review by ghost — DISMISSED** [review]
you finally unified the file names

File: code/modules/admin/player_panel.dm [hunk 5]
```diff
@@ -410,7 +410,8 @@
 				dat += "in [disk_loc.loc] at ([disk_loc.x], [disk_loc.y], [disk_loc.z])</td></tr>"
 			dat += "</table>"
 
-		if(get_antagonists(/datum/antagonist/rev))
+		var/list/revs = get_antagonists(/datum/antagonist/rev)
+		if(revs.len > 0)
```
Comment (MoreRobustThanYou) [inline code/modules/admin/player_panel.dm]: LAZYLEN?

**Comment by BeeSting12** [comment]
This broke revs, round 77438. Rev head could flash people but it said they were resistant to it. Rev head eventually got killed and round did not end, I ended it manually.

**Comment by Hatterhat** [comment]
DOESN'T FUCKING WORK REEE  EE EEEEE E EE E

**Comment by Hatterhat** [comment]
@AnturK FIX UR SHIDD

**Comment by BeeSting12** [comment]
https://file.house/eqDt.png

Rev targets are shown on all roundtypes, not just rev rounds.

**Comment by AnturK** [comment]
Already fixed.

**Pull request merged**